app/scripts/models/account.js

```diff
@@ -145,10 +145,30 @@ define([
       return this.get('verified') && ! this.get('accessToken');
     },
 
+    _generateAssertionPromise: null,
+    _generateAssertion: function () {
+      var self = this;
+
+      // share assertions between all concurrent assertion requesters. This
+      // eliminates multiple calls being made to sign certificates.
+      // See #3085
+      if (self._generateAssertionPromise) {
+        return self._generateAssertionPromise;
+      }
+
+      self._generateAssertionPromise =
```

Comment:
@zaach - I knew that was a corner case and went "meh." I'm trying to imagine a scenario where two two distinct session tokens would be used to create an assertion at the exact same time. It seems like if that happens, other issues are afoot. I can store by sessionToken though just to be safe, because really, who knows what'll happen in the future.